### PR TITLE
Support for nested relationship object when access the relationship

### PR DESCRIPTION
**Pull request by pitiphong-p**
If I have entity A->B->C when I call a.b the AFIS will fetching the b data from server but I find that AFIS won't set up the c relationship of B. I think it should.

I find that in 
(id)newValueForRelationship:(NSRelationshipDescription *)relationship forObjectWithID:(NSManagedObjectID *)objectID withContext:(NSManagedObjectContext *)context error:(NSError *__autoreleasing *)error doesn't set up the relationship.

### CODE REVIEW

File: AFIncrementalStore/AFIncrementalStore.m [hunk 1]
```diff
@@ -581,6 +581,58 @@ - (id)newValueForRelationship:(NSRelationshipDescription *)relationship
                         if (relationshipObjectID == nil) {
                             [childContext insertObject:managedRelationshipObject];
                         }
+
```
Comment (xslim) [inline AFIncrementalStore/AFIncrementalStore.m:584]: I think this method is already too long. Making it longer will make it more heavy, harder to read and to test.

**Comment by mattt** [comment]
Thanks for the pull request, @pitiphong-p. Correct me if I'm wrong, but it seems that this approach would only support 2 levels deep, no? 

A recursive solution would support any arbitrary depth, and even help to break up the wall of code in that method now. I've implemented that in 8743ea97c2599ee7edb1c2e1023218cf2583dcda ... e487c6a.

Let me know how that works for you. Cheers!

**Comment by pitiphong-p** [comment]
Yes you're right, mine approach support only 2 levels deep.

Your recursive solution is not work for me. 

```
- (id)newValueForRelationship:(NSRelationshipDescription *)relationship
              forObjectWithID:(NSManagedObjectID *)objectID
                  withContext:(NSManagedObjectContext *)context
                        error:(NSError *__autoreleasing *)error
```

This method is still not using your approach right?

**Comment by mattt** [comment]
Heh, almost forgot that one. Added in 34815bb.

> 1 files changed, 5 insertions(+), 39 deletions(-)

:+1:

**Comment by pitiphong-p** [comment]
I use the new 

```
- (void)insertOrUpdateObjectsFromRepresentations:(id)representationOrArrayOfRepresentations
                                        ofEntity:(NSEntityDescription *)entity
                                    fromResponse:(NSHTTPURLResponse *)response
                                     withContext:(NSManagedObjectContext *)context
                                           error:(NSError *__autoreleasing *)error
                                 completionBlock:(void (^)(NSArray *managedObjects, NSArray *backingObjects))completionBlock
```

in 

```
- (id)newValueForRelationship:(NSRelationshipDescription *)relationship
              forObjectWithID:(NSManagedObjectID *)objectID
                  withContext:(NSManagedObjectContext *)context
                        error:(NSError *__autoreleasing *)error
```

and it fixed my problem, also works for me too.

Should I send you pull request?

PS. Thanks for your great work on this project. I choose it in my new project too. :)

**Comment by pitiphong-p** [comment]
Oh just saw your commit.
Well it doesn't work I need to add 

```
                        NSManagedObjectContext *backingContext = [self backingManagedObjectContext];

                        NSManagedObject *managedObject = [childContext objectWithID:objectID];
                        NSManagedObject *backingObject = [backingContext existingObjectWithID:objectID error:nil];
                        if ([relationship isToMany]) {
                            if ([relationship isOrdered]) {
                                [managedObject setValue:[NSOrderedSet orderedSetWithArray:managedObjects] forKey:relationship.name];
                                [backingObject setValue:[NSOrderedSet orderedSetWithArray:managedObjects] forKey:relationship.name];
                            } else {
                                [managedObject setValue:[NSSet setWithArray:managedObjects] forKey:relationship.name];
                                [backingObject setValue:[NSSet setWithArray:managedObjects] forKey:relationship.name];
                            }
                        } else {
                            [managedObject setValue:[managedObjects lastObject] forKey:relationship.name];
                            [backingObject setValue:[backingObjects lastObject] forKey:relationship.name];
                        }
```

in the insertOrUpdate.. method completion block.

Your current implementation is just fetch and insert to the context. But not set the relationship back after inserted. So I had to manually set it back to the managed object.
Any idea?

**Comment by pitiphong-p** [comment]
Oh I found that current implementation is crash my program
I think you should use relationship.destinationEntity in ofEntity: argument?

**Comment by mattt** [comment]
Whoops, sorry about that. Fixed in 54f04e0.

**Comment by pitiphong-p** [comment]
Thanks :)
What's about the other problem?
I'm not sure that my solution is good enough.

**Comment by mattt** [comment]
@pitiphong-p Works for me :)

Just merged in your code in d7c7209. I committed it with you as the author, if that's alright with you. Thanks so much for your help on all of this! Sorry I wasn't hitting the same problems myself while debugging—I really need to come up with a more complex example to test against.

**Comment by pitiphong-p** [comment]
WOWWW That's great!

Thank you for your great work :)

**Pull request closed without merging**